src/Aspire.Hosting.Azure/AzureConstructResource.cs

```diff
@@ -109,6 +109,7 @@ public static IResourceBuilder<AzureConstructResource> AddAzureConstruct(this ID
     /// <param name="propertySelector">Property selection expression.</param>
     /// <param name="parameterResourceBuilder">Aspire parameter resource builder.</param>
     /// <param name="parameterName">The name of the parameter to be assigned.</param>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "<Pending>")]
     public static void AssignProperty<T>(this Resource<T> resource, Expression<Func<T, object?>> propertySelector, IResourceBuilder<ParameterResource> parameterResourceBuilder, string? parameterName = null) where T: notnull
```

Comment:
I'm not sure this is an issue. We have two variants of this overload, both take a different argument type as their second argument.

Reply:
.. we don't need to worry too much about breaking this API because its covered by experimental.